src/BSON/Javascript.c

```diff
@@ -227,6 +233,116 @@ PHP_METHOD(Javascript, getScope)
 }
 /* }}} */
 
+/* {{{ proto string Javascript::serialize()
+*/
+PHP_METHOD(Javascript, serialize)
+{
+	php_phongo_javascript_t  *intern;
+#if PHP_VERSION_ID >= 70000
+	zval                      retval;
+#else
+	zval                     *retval;
+#endif
+	php_phongo_bson_state state = PHONGO_BSON_STATE_INITIALIZER;
+
+	php_serialize_data_t      var_hash;
+	smart_str                 buf = { 0 };
+
+	intern = Z_JAVASCRIPT_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+#if PHP_VERSION_ID >= 70000
+	if (intern->scope && intern->scope->len) {
+		phongo_bson_to_zval_ex(bson_get_data(intern->scope), intern->scope->len, &state);
```

Comment:
This function can throw an exception and return zero. 

If you look at `php_phongo_javascript_get_properties()`, it stores null for the "scope" field (as if it didn't exist at all) and allows the function to return. That works out because nothing later in the function would throw. In this case, it might make more sense to return early instead of continuing to call `php_var_serialize()` below.